app.rb

```diff
@@ -0,0 +1,73 @@
+require 'sinatra/base'
+require 'sinatra'
+require 'sinatra/flash'
+require './lib/database_connection_setup'
+require './lib/tweets'
+require './lib/user'
+require './lib/comment'
+require 'uri'
+
+class Twitter < Sinatra::Base 
+  enable :sessions, :method_overwrite
+  register Sinatra::Flash
+
+  get '/' do 
+    erb :index
+  end 
+
+  get '/new_user' do
+    erb :sign_up
+  end 
+
+  post '/users' do 
+    user = User.create(name: params[:username], email: params[:email], password: params[:password])
+    session[:user_id] = user.id
+    redirect '/home'
+  end 
+
+  get '/log_in' do 
+    erb :log_in
+  end 
+
+  post '/user_log_in' do 
+    user = User.authenticate(name: params[:username], password: params[:password])
+    
+    if user 
+      session[:user_id] = user.id
+      redirect '/home'
+    else 
+      flash[:notice] = 'Please check your username or password.'
+      redirect '/log_in'
+    end 
+  end 
+
+  get '/home' do 
+    $user = User.find(session[:user_id])
+    @goliath = Tweets.all.reverse
+    erb :tweets
+  end 
+
```

Comment:
Could you perhaps use session parameters or instance variables to avoid using a global variable here?